### DISCUSSION TITLE
Need help finding documentation about the shortcode arguments format

### DISCUSSION THREAD

**Discussion by iusgit** [quarto-dev/quarto-cli#10434] [Q&A] [answered]
### Description

Hi,

it seems I had a deep sleep between Quarto version 1.3 and 1.5...

I have a shortcode that I used to print quantities in Latex with the siunitx package. It starts like:
```{lua}
function qty(args, kwargs, meta)
    print("In qty")
    quarto.log.output(args)
    quarto.log.output(kwargs)
    quarto.log.output(meta)
end
```
and would allow something like `{{ qty E13 \becquerel }}` in the QMD file.

In Quarto 1.3 the printout of `args` was like:
```
List {
  [1] {
    [1] Str {
      clone: function: 0x7fb5420e7120
      text: "E13"
      walk: function: 0x7fb5420e7460
    }
  }
  [2] {
    [1] RawInline {
      clone: function: 0x7fb5420e81c0
      format: "tex"
      text: "\becquerel "
      walk: function: 0x7fb5420e8e90
    }
  }
}
```

Now, in Quarto 1.4 and 1.5 the printout is:
```
In qty
List {[1] "1.0E+13", [2] "\becquerel"}
```

I used the additional information of "RawInline format:"tex"" to differentiate between normal text and Latex parts.

Can somebody tell me, if this a PANDOC or a QUARTO change?

Does anybody know where to find some documentation about this?

I thought the new fourth argument "raw_args" could give me the old format back, but it is not...

Thanks for any help!

**Comment by mcanouil** [comment]
I don't think the proper solution is to go backward, but to upgrade your shortcode LUA filter based on current Quarto/Pandoc API which is described in both tools documentation.

We could help with that if you provide the code, without there is not much they can be done.

**Comment by iusgit** [comment]
OK, so I would like to find out, whether I deal with a latex raw string in `args`. Before Quarto 1.4 I could question the "format" item. 
Like so:

```
function qty(args, kwargs, meta)
    for i, a in ipairs(args) do
        if a[1].format == 'tex' then
            print(a[1].text)
        end
    end
end
```

It seems that now strings are transfered by argument and this information is gone.

How do I get the same information in Quarto version >1.3 like "a[1].format"?

Thank you!

**Reply by mcanouil** [reply]
Could you provide all the pieces for us to be able to try your shortcode and help?

Providing only snippets of code without any example, etc. means we have to guess while you know exactly how it is supposed to work your shortcode.

**Reply by mcanouil** [reply]
````qmd
---
title: "Quarto Playground"
format:
  pdf:
    html-math-method: mathjax
    include-in-header:
      - text: |
          \usepackage{siunitx}
shortcodes: 
  - qty.lua
---


{{< qty E13 \becquerel >}}

$$
\qty{E13}{\becquerel}
$$
````

With `qty.lua` being:

```lua
return {
  ['qty'] = function(args)
    if quarto.doc.is_format("latex") then
      return pandoc.RawInline(
        'latex',
        '\\qty{' .. pandoc.utils.stringify(args[1]) .. '}' ..
        '{' .. pandoc.utils.stringify(args[2]) .. '}'
      )
    else
      return pandoc.Null()
    end
  end
}
```

I still have no idea what's your shortcode, how do you use it and what's exactly your issue but the code above works.

**Comment by iusgit** [comment, marked as answer]
Hello,

first of all a big thank you to @mcanouil for analyzing the situation!

After looking deeper into my code I saw that the old format of the `args` argument (until Quarto 1.3) was only used a few times in my shortcode code. 

So I found a way to keep compatibility to older Quarto versions by applying conditionals like so

```
if (quarto.version < '1.4.0' ) then
    if a[1].format then
        if a[1].format == 'tex' then
            val = a[1].text
        end    
    end
end
```
and using the new argument format that came with Quarto version 1.4.
